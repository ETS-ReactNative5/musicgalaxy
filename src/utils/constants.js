export const NAVIGATION_ROUTES = {
    HOME: 'Home',
    SEARCH: 'Search',
    ONBOARDING: 'Onboarding',
    VIEW_ALL: 'List',
    SEARCH: 'Search'
};

export const TEST_IDS = {
    FALLBACK_TITLE: 'FALLBACK_TITLE',
    FALLBACK_IMAGE_SRC: 'FALLBACK_IMAGE_SRC',
    LIST_CAROUSEL_ITEM_TITLE: 'LIST_CAROUSEL_ITEM_TITLE',
    LIST_CAROUSEL_ITEM_IMAGE_SRC: 'LIST_CAROUSEL_ITEM_IMAGE_SRC',
    LIST_CAROUSEL_ITEM_ARTIST_TEXT: 'LIST_CAROUSEL_ITEM_ARTIST_TEXT',
    LIST_CAROUSEL_ITEM_RELEASE_DATE: 'LIST_CAROUSEL_ITEM_RELEASE_DATE',
    LIST_CAROUSEL_HEADER_TITLE: 'LIST_CAROUSEL_HEADER_TITLE',
    LIST_CAROUSEL_SEE_ALL_TITLE: 'LIST_CAROUSEL_SEE_ALL_TITLE',
    LIST_CAROUSEL_FLATLIST: 'LIST_CAROUSEL_FLATLIST',
    TEXT_INPUT: 'TEXT_INPUT',
}

export const bannerImgSrc = 'https://images.unsplash.com/photo-1508700115892-45ecd05ae2ad?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=3269&q=80';

export const backIconSvg = `<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
<path d="M3.5 12L2.79289 11.2929C2.40237 11.6834 2.40237 12.3166 2.79289 12.7071L3.5 12ZM10.2071 17.2929L4.20711 11.2929L2.79289 12.7071L8.79289 18.7071L10.2071 17.2929ZM4.20711 12.7071L10.2071 6.70711L8.79289 5.29289L2.79289 11.2929L4.20711 12.7071ZM3.5 13L20.5 13L20.5 11L3.5 11L3.5 13Z" />
</svg>`;

export const searchIconSvg = `<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M16 10C16 13.3137 13.3137 16 10 16V18C14.4183 18 18 14.4183 18 10H16ZM10 16C6.68629 16 4 13.3137 4 10H2C2 14.4183 5.58172 18 10 18V16ZM4 10C4 6.68629 6.68629 4 10 4V2C5.58172 2 2 5.58172 2 10H4ZM10 4C13.3137 4 16 6.68629 16 10H18C18 5.58172 14.4183 2 10 2V4ZM14.2421 15.6574L20.5418 21.9571L21.956 20.5429L15.6563 14.2432L14.2421 15.6574Z"/>
</svg>`;
