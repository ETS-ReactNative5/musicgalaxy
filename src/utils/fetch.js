import axios from 'axios';

export const get = async (url, options, params) => {
  try {
    const response = await axios.get(
      `${url}${toQueryString(params)}`,
      options,
    );
    return response.data;
  } catch (error) {
    console.log('Error', error);
    throw error;
  }
};