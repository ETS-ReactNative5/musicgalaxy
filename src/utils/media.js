/* eslint-disable no-shadow */
/**
 * Problem: React Native accepts either percentages or
 * density-independent pixels (DP) for its styles.
 * Percentages work well on the web,
 * but they don't support all the style properties in React Native.
 * Only padding, margin, width, height, minWidth, minHeight,
 * maxWidth, maxHeight, flexBasis can have percentage values
 * according to this commit.
 * https://github.com/facebook/react-native/commit/3f49e743bea730907066677c7cbfbb1260677d11
 * Density independent pixels (DP) are different than traditional pixels.
 * The bigger the device, the more DP it has. However,
 * density-independent pixels (DP) vary from device to device,
 * so it cannot be directly used for creating a responsive layout.
 *
 * In order to make the app look good on phones, tablets,
 * and a variety of other devices, we need to scale all of our styles.
 *
 * And we use the following functions
 * scale, widthPercentageToDP and heightPercentageToDP
 */
 import {Dimensions, PixelRatio} from 'react-native';

 let {width: screenWidth, height: screenHeight} = Dimensions.get('window');
 
 export const SCREEN_WIDTH = screenWidth;
 export const SCREEN_HEIGHT = screenHeight;
 
 const widthScale = screenWidth / 414;
 const heightScale = screenHeight / 896;
 
 /**
    * Use scale() function to scale your fonts,
    * margins and paddings w.r.t screen width and height
    * @param {number} size
    * pixel value to scale w.r.t screen dpi
    * @param {string} based
    * the value with respect to width or height
    * based = 'width' | 'height' default 'width',
    * https://github.com/NewBieBR/react-native-normalize/blob/master/src/index.ts
    * @param {function} accessibilityScaleFactor
    * Func  to get Font size by PixelRatio.
    * if accessibility is > 100% (1) than multiple font size with PixelRatio.
    * if accessibility is > 150% (1.5) than divide font size with PixelRatio + 1.8.
 
    */
 export const scale = (size, based = 'width', accessibilityScaleFactor) => {
   let newSize = based === 'height' ? size * heightScale : size * widthScale;
   if (screenWidth < 768) {
     if (accessibilityScaleFactor) {
       newSize = accessibilityScaleFactor(newSize);
     }
     return Math.round(PixelRatio.roundToNearestPixel(newSize));
   }
   return size * 1.273;
 };
 
 export const scaleFont = size => {
   const fontScale = PixelRatio.getFontScale();
   return scale(size, 'width', size => {
     if (fontScale >= 1) {
       if (fontScale > 1.5) {
         return size / fontScale + 1.8;
       }
       return size * fontScale;
     }
     return size;
   });
 };
 