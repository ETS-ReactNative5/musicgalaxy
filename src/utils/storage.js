import EncryptedStorage from 'react-native-encrypted-storage';

/**
 * Loads something from storage and runs it thru JSON.parse.
 *
 * @param key The key to fetch.
 */
export const load = async (key) => {
    try {
        const value = await EncryptedStorage.getItem(key);
        return value;
    } catch {
        return null;
    }
}

/**
 * Saves an object to storage.
 *
 * @param key The key to fetch.
 * @param value The value to store.
 */
export const save = async (key, value) => {
    try {
        await EncryptedStorage.setItem(key, value);
        return true;
    } catch(err) {
        console.log('ERRROR',err)
        return false;
    }
}


/**
 * Burn it all to the ground.
 */
export const clear = async () => {
    try {
        await EncryptedStorage.clear();
        return true;
    } catch {
        return false;
    }
}

export default {
    clear,
    load,
    save,
};
