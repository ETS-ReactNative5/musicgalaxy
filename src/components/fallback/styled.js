import { APP_COLORS } from '@src/theme/colors';
import { scale } from '@src/utils/media';
import FastImage from 'react-native-fast-image';
import styled from 'styled-components/native';

export const Container = styled.View`
  flex: 1;
  background-color: ${APP_COLORS.white};
  justify-content: center;
  align-items: center;
`;

export const StyledImage = styled(FastImage)`
  height: 20%;
  width: 60%;
`;

export const TitleText = styled.Text`
  text-align: center;
  font-size: ${scale(16)}px;
  color: red;
  padding-horizontal: 10px;
`;
