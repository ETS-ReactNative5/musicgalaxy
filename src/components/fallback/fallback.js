import React from 'react';
import FastImage from 'react-native-fast-image';

import { TEST_IDS } from '@src/utils/constants';
import { Container, StyledImage, TitleText } from './styled';
import { propTypes, defaultProps } from './props';

/**
 * Component to show as a Fallback component in case of any faliure. Avoid App crashing
 *
 */
export const FallBackUI = () => {
    return (
        <Container>
            <StyledImage
                testID={TEST_IDS.FALLBACK_IMAGE_SRC}
                source={require('@assets/error.png')}
                resizeMode={FastImage.resizeMode.contain}
            />
            <TitleText testID={TEST_IDS.FALLBACK_TITLE}>
                Something went wrong while fetching the list of Music Videos :(
            </TitleText>
        </Container>
    );
};

FallBackUI.propTypes = propTypes;
FallBackUI.defaultProps = defaultProps;
