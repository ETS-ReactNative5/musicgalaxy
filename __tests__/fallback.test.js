import React from 'react';

import { FallBackUI } from '@src/components/fallback/fallback';
import { render } from '@testing-library/react-native';
import { TEST_IDS } from '@src/utils/constants';

const imgSource = require('@assets/error.png');

describe('Testing FallBack Component', () => {

    it('should match the Fallback component snapshot', () => {
        const { toJSON } = render(<FallBackUI />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('should render Fallback error message correctly', () => {
        const { getByTestId } = render(<FallBackUI />);
        expect(getByTestId(TEST_IDS.FALLBACK_TITLE).props.children).toBe('Something went wrong while fetching the list of Music Videos :(');
    });

    it('should render Fallback error image source correctly', () => {
        const { queryByTestId } = render(<FallBackUI />);
        expect(queryByTestId(TEST_IDS.FALLBACK_IMAGE_SRC).props.source).toBe(imgSource);
    });

});

