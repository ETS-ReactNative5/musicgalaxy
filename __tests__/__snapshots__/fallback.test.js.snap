// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FallBack Component should match the Fallback component snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Array [
          Object {
            "height": "20%",
            "width": "60%",
          },
        ],
      ]
    }
  >
    <FastImageView
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../assets/error.png",
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="FALLBACK_IMAGE_SRC"
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "red",
          "fontSize": 29,
          "paddingHorizontal": 10,
          "textAlign": "center",
        },
      ]
    }
    testID="FALLBACK_TITLE"
  >
    Something went wrong while fetching the list of Music Videos :(
  </Text>
</View>
`;
